### PR TITLE
updating duckduckgo version

### PR DESCRIPTION
**Pull request by zedatrix**
During testing for the v0.5.0 release, we discovered web search was not working as expected. This was due to an old version of the sdk being used. This pull request just updates the version to 4 (currently the latest) and also changes the forge to local to use the changes. We should either add this to the docs or leave it default to local i think.

### CODE REVIEW

File: autogpts/autogpt/pyproject.toml [hunk 1]
```diff
@@ -22,16 +22,17 @@ serve = "autogpt.app.cli:serve"
 
 [tool.poetry.dependencies]
 python = "^3.10"
-# autogpt-forge = { path = "../forge" }
-autogpt-forge = {git = "https://github.com/Significant-Gravitas/AutoGPT.git", subdirectory = "autogpts/forge"}
+#TODO: Add this to the documentation so people can use the local or remote version. E.g. setting to local was the only way to fix the duck duck go dependency issue.
+autogpt-forge = { path = "../forge" }
+#autogpt-forge = {git = "https://github.com/Significant-Gravitas/AutoGPT.git", subdirectory = "autogpts/forge"}
```
Comment (Pwuts) [inline autogpts/autogpt/pyproject.toml:27]: This will break the Docker build.

**Review by Pwuts — CHANGES_REQUESTED** [review]

File: autogpts/autogpt/pyproject.toml (hunk 1 above)
Comment (zedatrix) [reply autogpts/autogpt/pyproject.toml:27]: Oh because it needs a remote repo.. ok I wanted to make it a flag anyways.. I can add that maybe?

**Comment by Pwuts** [comment]
The proper way to do this would be:
1. Update `duckduckgo-search` in `forge`
2. Update `forge` and `duckduckgo-search` in `autogpt`

This has to happen in 2 separate commits, because `autogpt/poetry.lock` needs to reference a specific commit for its `forge` dependency.

File: autogpts/autogpt/pyproject.toml (hunk 1 above)
Comment (zedatrix) [reply autogpts/autogpt/pyproject.toml:27]: Or maybe the Dockerfile file can account for it?

**Comment by Pwuts** [comment]
There was a bit of time pressure on this, so I adopted your fix in d820239a. Thanks for figuring it out! :)

**Pull request closed without merging**